cpu/native/socket_zep/socket_zep.c

```diff
@@ -0,0 +1,441 @@
+        thread_yield();
+    }
+    res = writev(dev->sock_fd, v, n + 2);
+    if (res < 0) {
+        DEBUG("socket_zep::send: error writing packet: %s\n", strerror(errno));
+        return res;
+    }
+    /* simulate TX_COMPLETE interrupt */
+    if (netdev->event_callback) {
+        dev->last_event = NETDEV_EVENT_TX_COMPLETE;
+        netdev->event_callback(netdev, NETDEV_EVENT_ISR);
+        thread_yield();
+    }
+#ifdef MODULE_NETSTATS_L2
+    netdev->stats.tx_bytes += bytes;
+#else
+    (void)bytes;
+#endif
+
+    return res - v[0].iov_len - v[n + 1].iov_len;
+}
+
+static void _continue_reading(socket_zep_t *dev)
+{
+    /* work around lost signals */
+    fd_set rfds;
+    struct timeval t;
+    memset(&t, 0, sizeof(t));
+    FD_ZERO(&rfds);
+    FD_SET(dev->sock_fd, &rfds);
+
+    _native_in_syscall++; /* no switching here */
+
+    if (real_select(dev->sock_fd + 1, &rfds, NULL, NULL, &t) == 1) {
+        int sig = SIGIO;
+        extern int _sig_pipefd[2];
+        extern ssize_t (*real_write)(int fd, const void * buf, size_t count);
+        real_write(_sig_pipefd[1], &sig, sizeof(int));
+        _native_sigpend++;
+    }
+    else {
+        native_async_read_continue(dev->sock_fd);
+    }
+
+    _native_in_syscall--;
+}
+
+static inline bool _dst_not_me(socket_zep_t *dev, const void *buf)
+{
+    uint8_t dst_addr[IEEE802154_LONG_ADDRESS_LEN] = { 0 };
+    int dst_len;
+    le_uint16_t dst_pan = { .u16 = 0 };
+
+    dst_len = ieee802154_get_dst(buf, dst_addr,
+                                 &dst_pan);
+    switch (dst_len) {
+        case IEEE802154_LONG_ADDRESS_LEN:
+            return memcmp(dst_addr, dev->netdev.long_addr, dst_len) != 0;
+        case IEEE802154_SHORT_ADDRESS_LEN:
+            return (memcmp(dst_addr, ieee802154_addr_bcast, dst_len) != 0) &&
+                   (memcmp(dst_addr, dev->netdev.short_addr, dst_len) != 0);
+        default:
+            return false;    /* better safe than sorry ;-) */
+    }
+}
+
+static int _recv(netdev_t *netdev, void *buf, size_t len, void *info)
+{
+    socket_zep_t *dev = (socket_zep_t *)netdev;
+    int size = 0;
+
+    DEBUG("socket_zep::recv(%p, %p, %u, %p)\n", (void *)netdev, buf,
+          (unsigned)len, (void *)info);
+    if ((buf == NULL) || (len == 0)) {
+        int res = real_ioctl(dev->sock_fd, FIONREAD, &size);
+#if ENABLE_DEBUG
+        if (res < 0) {
+            DEBUG("socket_zep::recv: error reading FIONREAD: %s",
+                  strerror(errno));
+        }
+#else
+        (void)res;
+#endif
+        return size;
+    }
+    else if (len > 0) {
+        size = real_read(dev->sock_fd, dev->rcv_buf, sizeof(dev->rcv_buf));
+
+        if (size > 0) {
+            zep_hdr_t *tmp = (zep_hdr_t *)&dev->rcv_buf;
+
+            if ((tmp->preamble[0] != 'E') || (tmp->preamble[1] != 'X')) {
+                DEBUG("socket_zep::recv: invalid ZEP header");
+                return -1;
+            }
+            switch (tmp->version) {
```

Comment:
guess https://github.com/RIOT-OS/RIOT/pull/6121#discussion_r119191851 invalidates this comment?